src/pkg/services/m365/api/channels_pager.go

```diff
@@ -2,38 +2,143 @@ package api
 
 import (
 	"context"
+
```

Comment:
This PR is already getting a little to large to fully review.  Can you please move the pagers implementation into a follow-up PR?  In fact, any addition of new code on top of the original PR can be moved to a follow-up PR, for ease of review.  We can flatten them all before merging to main if extra assurance is needed.